### PR TITLE
Add architecture docs for Search Service

### PR DESCRIPTION
**Pull request by nshumoogum**
### What

Architecture docs contains the following:
- link to google drive for c4 diagrams
- several sequence diagrams
- further details for each step in sequence

### How to review

Check documentation makes sense (this was adapted from original sequence diagram before work started)

### Who can review

Anyone

### CODE REVIEW

File: architecture/README.md [hunk 1]
```diff
@@ -0,0 +1,654 @@
+# Search Service Architecture
+
+Source of truth for the application architecture of the ONS search service including backend processing of data. C4 diagrams available via [ONS google drive](https://drive.google.com/drive/folders/1usE1C-VzQIrymSJ7tJzXxXjZAmXLge2S).
```
Comment (lindenmckenzie) [inline architecture/README.md:3]: This is the old google drive Nathan

Comment (lindenmckenzie) [reply architecture/README.md:3]: Correct one here: https://drive.google.com/drive/folders/15Qq3wgaULer96kXGuMDUo6D0-vAz7um1

File: architecture/README.md [hunk 2]
```diff
@@ -0,0 +1,654 @@
...
+BSON Document update: {
+    "last_updated": ISODate, // ISO8601 timestamp
+    "reindex_failed": ISODate,
+    "state": "failed"
+}
+```
+
+## Sequence diagrams
+
+To update the diagrams, see [sequence diagrams documnetation](sequence-diagrams/README.md)
```
Comment (lindenmckenzie) [inline architecture/README.md:654]: *documentation

File: architecture/README.md [hunk 3]
```diff
@@ -0,0 +1,654 @@
...
+
+See [full schema here](https://github.com/ONSdigital/dp-search-data-extractor/blob/develop/schema/schema.go#L25)
+
+**7: Consume event from search-data-import**
+
+Kafka message consumed by Search Data Importer. Documents received via consumption of kafka topic `search-data-import`, store documents in memory until 500 messages consumed or time from first message exceeded 5 seconds before making bulk request. 5 second limit will allow for the last set of messages to still be reindexed.
+
+**Note: For search reindex process, we create separate batches based on the search index name; this way we can insert documents to the correct index for both reindex jobs and publishing new content and data**
+
+**8: Multi-operational request to update/add docs to an index**
```
Comment (lindenmckenzie) [inline architecture/README.md:118]: why do we connect direct to ES here? just interested in the answer really.

File: architecture/README.md [hunk 4]
```diff
@@ -0,0 +1,654 @@
...
+
+If the batch of documents was for a reindex job, then produce event for the search reindex tracker.
+
+```
+Datastore: kafka
+
+Topic: search-data-imported
+Record: {
+    "job_id": string, // mandatory
+    "number_of_search_documents // mandatory
```
Comment (lindenmckenzie) [inline architecture/README.md:145]: Missing closing quote mark.

File: architecture/README.md [hunk 5]
```diff
@@ -0,0 +1,654 @@
...
+Record: {
+    "uri": string, // mandatory
+    "search_index": string, // mandatory
+    "data_type": string, // mandatory
+    "collection_id": string, // optional
+    "job_id": string // optional (used for reindex pipeline)
+}
+```
+
+The `search_index` field should be set to the search index alias, `ONS` for publishing new content. Only search reindexes will make use of the actual index name, e.g. `ons_<timestamp>`.
```
[the remark this answers is not in the document]
Comment (nshumoogum) [reply architecture/README.md:67]: It actually doesn't matter, it is defined as `ons` in search API and then `ONS` in search data extractor. I'll create a PR to synch the search data extractor to be `ons` and update this here to `ons` too

File: architecture/README.md (hunk 3 above)
Comment (nshumoogum) [reply architecture/README.md:118]: So the only other place that talks to ES is the Search API - but I generally see the search data pipeline as separate entity to the Search API and hence didn't want to use it. Other more minor reasons, the importer may change a lot and didn't want to be restricted by backwards compatibility or versioning by using a new endpoint on the search API; less latency. The search data pipeline is only hosted in publishing subnet, less chance of exposing ability for users to write to database - even though this is the situation via other DP APIs.

Comment (nshumoogum) [reply architecture/README.md:118]: Generally it seemed to make sense to do it this way

**Review by lindenmckenzie — DISMISSED** [review]

**Review by lindenmckenzie — APPROVED** [review]

**Pull request merged**